CST338/Seperate files/assig3.java

```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
       }
       System.out.println("End of dealing shuffled cards");
       
+      System.out.println("\n--------Deck & Hand Interaction--------");
+      System.out.println("------Unshuffled Deck------");
+      // Declare P4 variables
+      int numPlayers, numCards;
       
+      final int MIN_PLAYERS = 1;
+      final int MAX_PLAYERS = 10;
+
+      boolean validInput = false;
+      Scanner userInput = new Scanner(System.in);  // Instantiate scanner object
+      
+      do
+      {
+         // Request user input
+         System.out.print("Enter number of players in game (1-10): ");
+         numPlayers = userInput.nextInt();
+
+         // Verify input
+         if (numPlayers >= MIN_PLAYERS && numPlayers <= MAX_PLAYERS)
+            
+            // Break loop if user input is good
+            validInput = true;
+
+      } while (!validInput);
       
+      // Instantiate game objects
+      Deck deck = new Deck();                   // Single deck, not shuffled
+      Hand players[] = new Hand[numPlayers];    // Array containing player hands
+      
+      // Give each player a hand
+      for (int i = 0; i < numPlayers; i++)
+         players[i] = new Hand();
+      
+      // Get number of cards in deck and store it
+      numCards = deck.topCard();                
+
+      // Deal cards to players
```

Comment:
all comments in loop should be here, block comment if necessary

Reply:
Sure thing